components/CraftItemRequirementsListInput.tsx

```diff
@@ -0,0 +1,292 @@
+import { CloseIcon } from '@chakra-ui/icons';
+import {
+  Button,
+  FormControl,
+  FormHelperText,
+  Grid,
+  HStack,
+  Image,
+  Input,
+  Text,
+  Tooltip,
+  VStack,
+} from '@chakra-ui/react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { useAccount } from 'wagmi';
+
+import { useGame } from '@/contexts/GameContext';
+import { CraftRequirement, Item } from '@/utils/types';
+
+import { SelectItemInput } from './SelectItemInput';
+
+type Props = {
+  craftRequirementsList: Array<CraftRequirement>;
+  setCraftRequirementsList: React.Dispatch<
+    React.SetStateAction<Array<CraftRequirement>>
+  >;
+};
+
+export const CraftItemRequirementsListInput: React.FC<Props> = ({
+  craftRequirementsList: requirementsList,
+  setCraftRequirementsList: setRequirementsList,
+}) => {
+  const { game } = useGame();
+  const { chain } = useAccount();
+
+  const { items, chainId } = game || {
+    items: [],
+    chainId: chain?.id,
+  };
+
+  const itemMap = useMemo(() => {
+    const map = new Map<string, Item>();
+    items.forEach(c => map.set(c.itemId, c));
+    return map;
+  }, [items]);
+
+  const removeCraftItemRequirement = useCallback(
+    (index: number) => {
+      const newCraftItemRequirementList = requirementsList.slice();
+      newCraftItemRequirementList.splice(index, 1);
+      setRequirementsList(newCraftItemRequirementList);
+    },
+    [requirementsList, setRequirementsList],
+  );
+
+  const itemsNotSelected = useMemo(() => {
+    const selectedAddresses = new Set(
+      requirementsList.map(c => c.itemId.toString()),
+    );
+    return items.filter(c => !selectedAddresses.has(c.itemId));
+  }, [requirementsList, items]);
+
+  if (!chainId) {
+    return (
+      <VStack align="stretch" spacing={4} w="100%">
+        <Text>You must connect your wallet to add claimers.</Text>
```

Comment:
Should this refer to "claimers" here? Seems like it should say something like "...to add requirements"

Reply:
fixed in [846ae47](https://github.com/raid-guild/character-sheets-frontend/pull/170/commits/846ae471e0454be0c7b863960a53c5166866b938)